src/ol/style/fill.js

```diff
@@ -31,6 +31,18 @@ ol.style.Fill = function(opt_options) {
 
 
 /**
+ * Clones the style.
+ * @return {ol.style.Fill} The cloned style.
+ * @api
+ */
+ol.style.Fill.prototype.clone = function() {
+  return new ol.style.Fill({
+    color: (this.getColor() instanceof Array) ? this.getColor().slice(0) : this.getColor()
```

Comment:
if i would use slice anyways to shallow copy an array, i could use this simplified variant:

```
ol.color.clone = function(color) {
  if (color.slice) {
    return color.slice();
  } else {
    return color;
  }
};
```